lib/rubocop/cop/packaging/gemspec_git.rb

```diff
@@ -1,43 +1,57 @@
 # frozen_string_literal: true
 
+# Rubocop module
 module RuboCop
+  # Cop module
   module Cop
+    # Packaging module
     module Packaging
-      # TODO: Write cop description and example of bad / good code. For every
-      # `SupportedStyle` and unique configuration, there needs to be examples.
-      # Examples must have valid Ruby syntax. Do not use upticks.
+      # This cop is used to identify the usage of `git ls-files`
+      # and suggests to use a plain Ruby alternative, like `Dir`,
+      # `Dir.glob` or `Rake::FileList` instead.
       #
-      # @example EnforcedStyle: bar (default)
-      #   # Description of the `bar` style.
+      # @example
       #
       #   # bad
-      #   bad_bar_method
+      #   Gem::Specification.new do |spec|
+      #     spec.files = `git ls-files`.split('\n')
+      #   end
       #
       #   # bad
-      #   bad_bar_method(args)
-      #
-      #   # good
-      #   good_bar_method
-      #
-      #   # good
-      #   good_bar_method(args)
-      #
-      # @example EnforcedStyle: foo
-      #   # Description of the `foo` style.
+      #   Gem::Specification.new do |spec|
+      #     spec.files = Dir.chdir(File.expand_path('..', __FILE__)) do
+      #       `git ls-files -z`.split('\\x0').reject { |f| f.match(%r{^(test|spec|features)/}) }
+      #     end
+      #   end
       #
       #   # bad
-      #   bad_foo_method
+      #   Gem::Specification.new do |spec|
+      #     spec.files         = `git ls-files`.split('\n')
+      #     spec.test_files    = `git ls-files -- test/{functional,unit}/*`.split('\n')
+      #     spec.executables   = `git ls-files -- bin/*`.split('\n').map{ |f| File.basename(f) }
+      #   end
       #
-      #   # bad
-      #   bad_foo_method(args)
+      #   # good
+      #   Gem::Specification.new do |spec|
+      #     spec.files         = Dir['lib/**/*', 'LICENSE', 'README.md']
+      #     spec.test_files    = Dir['spec/**/*']
+      #   end
       #
       #   # good
-      #   good_foo_method
+      #   Gem::Specification.new do |spec|
+      #     spec.files         = Rake::FileList['**/*'].exclude(*File.read('.gitignore').split)
+      #   end
       #
       #   # good
-      #   good_foo_method(args)
+      #   Gem::Specification.new do |spec|
+      #     spec.files         = Dir.glob('lib/**/*')
+      #     spec.test_files    = Dir.glob('test/{functional,test}/*')
+      #     spec.executables   = Dir.glob('bin/*').map{ |f| File.basename(f) }
+      #   end
```

Comment:
this seems to duplicate the contents of `docs/modules/ROOT/pages/cops_packaging.adoc`. is this duplication really necessary?

Reply:
`docs/modules/ROOT/pages/cops_packaging.adoc` is generated from the document `lib/rubocop/cop/packaging/gemspec_git.rb` via a rake task.

Reply:
is there a reason to include generated files in the repository?

Reply:
Yes, I searched and saw that all the RuboCop extensions are using `generate_cops_documentation` task to generate documentation that maps to their respective docs. For example:
- https://docs.rubocop.org/rubocop-ast/
- https://docs.rubocop.org/rubocop-performance/
- https://docs.rubocop.org/rubocop-rspec/
- https://docs.rubocop.org/rubocop-minitest/
- and so on..

So they all have `docs/` directory which contains that documentation.
So if someone wants to add, fix, or remove something, that's the place they'll be wanting to look for.

Reply:
I believe the source of truth of docs is the `docs/` folder, _except for_ cops documentation. For those, the source of truth is the code comments inside the cops code. Given that, I think it makes sense to keep the generated `.adoc` files for cops documentation in source control to simplify the documentation site generation. I guess those could be generated on the fly, but sounds tricky.

Reply:
Well, now that I look at the last commit, I see that also the `index.adoc` is generated from the README. In any case, I would stick to the way it's done in general in the rubocop ecosystem.

Reply:
The pages that are coming out of README.md are:
- index.adoc
- installation.adoc
- usage.adoc

None of them are generated. Even for the RuboCop ecosystem. At the moment, I did them manually.
So the idea is that not all of the code can go into README, therefore, there exists docs/ separately which documents the important stuff.
For now, there's not much because it's just the start, otherwise, I'll keep documenting as we move forward and if somebody requests more stuff to be explained via docs.

Also for the cops documentation, most of it comes from the code comments but there are things like "references" or some extra links that can be added for better explanation and stuff.
For examle: https://github.com/rubocop-hq/rubocop-performance/blob/master/docs/modules/ROOT/pages/cops_performance.adoc#references
This is not a part of the code comments but it is there in the docs :)

Reply:
Ah, ok :+1:.

Reply:
@terceiro, do you want to add anything? Or are there any objects from your side?
(just a friendly ping, no rush -- we're waiting for your ack to get this merged.)

Once this is merged, I'll fix the gemspec file and then we're ready for an initial release :tada:

Reply:
nah just go ahead